runtime/sys.c

```diff
@@ -505,6 +505,54 @@ void caml_sys_init(char_os * exe_name, char_os **argv)
 #endif
 #endif
 
+#ifdef __ANDROID__
+#include <pthread.h>
+#include <spawn.h>
+
+/* Android 8 Oreo and later seem to block system(3). Provide a
+ * replacement, taken from musl.
+ * SPDX-License-Identifier: MIT
+ * Copyright (c) 2005-2020 Rich Felker, et al.
+ * https://git.musl-libc.org/cgit/musl/tree/src/process/system.c?h=v1.2.5 */
+int caml_sys_system(const char *cmd)
+{
+  pid_t pid;
+  sigset_t old, reset;
+  struct sigaction sa = { .sa_handler = SIG_IGN }, oldint, oldquit;
+  int status = -1, ret;
+  posix_spawnattr_t attr;
+
+  /* Bionic does not provide thread cancellation */
+  /* pthread_testcancel(); */
+
+  if (!cmd) return 1;
+
+  sigaction(SIGINT, &sa, &oldint);
+  sigaction(SIGQUIT, &sa, &oldquit);
+  sigaddset(&sa.sa_mask, SIGCHLD);
+  sigprocmask(SIG_BLOCK, &sa.sa_mask, &old);
+
+  sigemptyset(&reset);
+  if (oldint.sa_handler != SIG_IGN) sigaddset(&reset, SIGINT);
+  if (oldquit.sa_handler != SIG_IGN) sigaddset(&reset, SIGQUIT);
+  posix_spawnattr_init(&attr);
+  posix_spawnattr_setsigmask(&attr, &old);
+  posix_spawnattr_setsigdefault(&attr, &reset);
+  posix_spawnattr_setflags(&attr, POSIX_SPAWN_SETSIGDEF|POSIX_SPAWN_SETSIGMASK);
+  ret = posix_spawn(&pid, "/bin/sh", 0, &attr,
```

Comment:
Have you been able to check that `posix_spawn` is available on Android? Note that `configure` checks for its availability.

Reply:
Yes, it's been available since Android 8 Oreo.